test/utility/misc.c

```diff
@@ -358,6 +358,29 @@ testutil_print_command_line(int argc, char *const *argv)
     printf("\n");
 }
 
+/*
+ * testutil_wiredtiger_open --
+ *     Call wiredtiger_open with the tiered storage configuration if enabled.
+ */
+void
+testutil_wiredtiger_open(TEST_OPTS *opts, const char *config, WT_EVENT_HANDLER *event_handler,
+  WT_CONNECTION **connectionp, bool recovery)
+{
+    char buf[1024];
+
+    strncpy(buf, config, sizeof(buf));
+    if (recovery)
+        strncat(buf, TESTUTIL_ENV_CONFIG_REC, sizeof(buf) - strlen(buf) - 1);
```

Comment:
For all these strncat calls, do:
```
strncat(buf, TESTUTIL_ENV_CONFIG_REC, sizeof(buf));
```
Take a look at the strncat documentation and note that you are passing `buf` as the first argument, that is, the beginning of the buffer, not the current end of the string in the buffer (`buf + strlen(buf)`).

What you have will work, and is safe, but if prevents the full buffer size from being used (if needed).  But it's unusual and experienced code readers will wonder what's going on that they aren't seeing.

Reply:
Hey Ravi, let's use your example, and let's also assign a memory location for buf:  1000 (decimal).
```
char buf[15];
char numbers[]="0123456789";

/* Initialize buf */
buf[0]='\0';
```
location 1000 is assigned 0.
```
/* At first I append 10 characters to it and I pass the last argument sizeof(buf) - strlen(buf) == 10-0 */
strncat(buf, numbers, sizeof(buf) - strlen(buf));
```
location 1000 has '0'
location 1001 has '1'
etc.
location 1009 has '9'
```
/* Next time when I call strncat I should pass the available/leftover space in the buf which is 5(sizeof(buf) - strlen(buf) == 15-10=5) not 15 (sizeof(buf = 15))*/
```
Yes that's true, but:
```
strncat(buf, numbers, sizeof(buf) - strlen(buf));
```
This tells strncat that there is a buffer at 1000 that is length 5 (that is, the buffer goes from 1000-1004).  Strncat concatenates to what's there, so it will find the end of the string (starting at the beginning of that buffer and append to it.  It goes from 1000 to 1004 looking for the end of the string and then stops - because it was told the buffer is 5 bytes big.  There's no room to append anything, or even get the end of the existing string.

This is why the idiom is typically:
```
strncpy(buf, ...., sizeof(buf));
strncat(buf, ...., sizeof(buf);
strncat(buf, ...., sizeof(buf);
strncat(buf, ...., sizeof(buf);
....
```
If there are lots of strncat, though, we often use a single snprintf.

Reply:
Thanks @ddanderson for the details, but I see a different result when I run a similar example locally. Anyways I have updated the code with the suggested approach.

Reply:
Just to followup here in case anyone digs into this conversation.  Ravi is right, and I am wrong.  I'll followup with a ticket to correct the use of strncat that I suggested.  (I think snprintf or __wt_snprintf will be a cleaner approach FWIW).